src/main/java/com/amazonaws/codepipeline/jenkinsplugin/AWSCodePipelinePublisher.java

```diff
@@ -119,6 +125,11 @@ else if (model.getActionTypeCategory() == CategoryType.Test) {
         try {
             LoggingHelper.log(listener, "Publishing artifacts");
 
+            /* TODO should this restriction remain?
+
+            Maybe this should not be a requirement - a build should be
+            allowed to output more artifacts then a pipeline may need.
+
```

Comment:
I think we still need this restriction:

* If `# of outputs in pipeline < # of outputs in Jenkins`
    * Each output artifact tells you to what S3 location (bucket/key) to upload. The plugin wouldn't know where to upload the artifacts that CodePipeline is unaware of.
    * There's nothing preventing your project from producing more outputs, but for outputs that CodePipeline is unaware of, I think it's best to use other Jenkins plugins, like the S3 Plugin.

* If `# of outputs in pipeline > # of outputs in Jenkins`
    * The next actions in the pipeline can use any Jenkins output as an input. If we don't produce an output for it, the action(s) that depend on it will likely fail.